File: src/main/java/me/centralhardware/znatoki/telegram/statistic/telegram/CommandHandler/StartCommand.java
package me.centralhardware.znatoki.telegram.statistic.telegram.CommandHandler;

import lombok.RequiredArgsConstructor;
import me.centralhardware.znatoki.telegram.statistic.Config;
import me.centralhardware.znatoki.telegram.statistic.entity.postgres.Role;
import me.centralhardware.znatoki.telegram.statistic.mapper.postgres.OrganizationMapper;
import org.springframework.stereotype.Component;
import org.telegram.telegrambots.meta.api.objects.Message;

@Component
@RequiredArgsConstructor
public class StartCommand extends CommandHandler {

    private final OrganizationMapper organizationMapper;

    @Override
    public void handle(Message message) {
        sender.sendText("""
                        Бот предназначенный для анализа вашего бизнеса.
                        Автор: @centralhardware
                        """, message.getFrom());
        if (!organizationMapper.exist(message.getFrom().getId())){
            sender.sendText(Config.Telegram.getStartTelegraph(), message.getFrom());
        }
    }

    @Override
    public boolean isAcceptable(String data) {
        return data.equals("/start");
    }

    @Override
    public Role getRequiredRole() {
        return null;
    }

}

File: src/main/kotlin/me/centralhardware/znatoki/telegram/statistic/telegram/commandHandler/StartCommand.kt
package me.centralhardware.znatoki.telegram.statistic.telegram.commandHandler

import dev.inmo.tgbotapi.extensions.api.send.sendMessage
import dev.inmo.tgbotapi.extensions.behaviour_builder.BehaviourContext
import dev.inmo.tgbotapi.extensions.behaviour_builder.triggers_handling.onCommand

fun BehaviourContext.startCommand() = onCommand("start") {
    bot.sendMessage(
        it.chat,
        """
                         Бот предназначенный для анализа вашего бизнеса.
                         Автор: @centralhardware
                         """
            .trimIndent(),
    )
}
